zkvm/src/vm.rs

```diff
@@ -736,6 +754,13 @@ where
             .map(|(content, _)| content)
     }
 
+    fn set_program(&mut self, prog: Data) -> Result<(), VMError> {
```

Comment:
Nice! How about a more accurate name, something like:

```suggestion
    fn continue_with_program(&mut self, prog: Data) -> Result<(), VMError> {
```